Feature/task 80 nikonov

DenNikonov opened the pull request:
Создан интерфейс для работы с дисциплинами.

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создать интерфейс для работы с дисциплинами

Создан интерфейс для работы с дисциплинами.

DenNikonov committed:
Merge branch 'master' into feature/Task_80_Nikonov

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создать интерфейс для работы с дисциплинами

Создан интерфейс для работы с дисциплинами

DenNikonov committed:
ПЕ-21б Никонов Д.О. Создать интерфейс для работы с дисциплинами

Добавлены представления и контроллер в проект
Добавлена проверка на доступ
Изменен внешний вид и скрипт на удаление

DenNikonov committed:
Merge branch 'master' into feature/Task_80_Nikonov

# Conflicts:
#	IS.Model/IS.Model.Tests/production.App.config
#	IS.Mvc/IS.Mvc/IS.Mvc.csproj

DenNikonov committed:
ПЕ-21б Никонов Д.О Создать интерфейс для работы с дисциплинами

добавлен продакшн

DenNikonov committed:
ПЕ-21б Никонов Создать интерфейс для работы с дисциплинами

Добавлен SQL/Task_80/Insert into Access.role values creator,deleter,updater,reader.sql

dha01 commented on SQL/Task_80/Insert into Access.role values creator,deleter,updater,reader.sql:
```diff
@@ -0,0 +1,10 @@
+           (code
+           ,mem)
+     values
+           ('Discipline.Creator','��������� ������� ����������'),
```
Кодировка.

dha01 commented on IS.Mvc/IS.Mvc/production.Web.config:
```diff
@@ -9,7 +9,7 @@
     <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=4.4.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
   </configSections>
   <connectionStrings>
-    <add name="DefaultConnection" providerName="System.Data.SqlClient" connectionString="Data Source=DHA-PC\SQLEXPRESS; Initial Catalog=IS; User Id=sa; Password=12345" />  </connectionStrings>
```
Не надо коммитить изменения тут.

dha01 commented on IS.Model/IS.Model.Tests/production.App.config:
```diff
@@ -5,7 +5,6 @@
   <connectionStrings>
-	<add name="DefaultConnection" providerName="System.Data.SqlClient" connectionString="Data Source=DHA-PC\SQLEXPRESS; Initial Catalog=IS_backup; Integrated Security=SSPI" />
-  </connectionStrings>
+	<add name="DefaultConnection" providerName="System.Data.SqlClient" connectionString="Data Source=DHA-PC\SQLEXPRESS;Initial Catalog=IS_backup; Initial Catalog=IS; Integrated Security=SSPI" />  </connectionStrings>
```
Не надо коммитить изменения тут.

dha01 commented on IS.Mvc/IS.Mvc/Controllers/TaskController.cs:
```diff
@@ -82,6 +82,10 @@ public ActionResult Update(TaskItem task)
 		{
 			Access.CheckAccess("Task.Updater");
 			_taskService.Update(task);
+			var _disciplineService = new DisciplineService();
```
Не надо коммитить всё подряд.
Нужно после коммито проверить что ничего лишнего нет.
Revert в студию.

dha01 commented on IS.Mvc/IS.Mvc/Controllers/DisciplineController.cs:
```diff
@@ -0,0 +1,116 @@
+	{
+		private DisciplineService _disciplineService;
+
+		public DisciplineController()
```
Добавить комментарий.
Можно написать, что тут создается сервис

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создать интерфейс для работы с дисциплинами

Откат изменений в WebConfig и appConfig
Изменения во внешнем виде
Исправлена кодировка в файле sql запроса

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создать интерфейс для работы с дисциплинами

Реверт TaskController
Комментарий в DisciplineController

dha01 commented on IS.Mvc/IS.Mvc/Controllers/DisciplineController.cs:
```diff
@@ -0,0 +1,118 @@
+		private DisciplineService _disciplineService;
+    /// <summary>
+    /// Создание сервиса дисциплин.
+    /// </summary>
```
Пробелы...

dha01 commented on IS.Mvc/IS.Mvc/Controllers/DisciplineController.cs:
```diff
@@ -0,0 +1,118 @@
+		}
+
+		/// <summary>
+		/// Интерфейс для создания задачи.
```
Задачи?

dha01 commented on IS.Mvc/IS.Mvc/Controllers/DisciplineController.cs:
```diff
@@ -0,0 +1,118 @@
+			Access.CheckAccess("Discipline.Creator");
+			var defaultitem = new DisciplineItem()
+			{
+				ShortName = "Краткое ",
```
Тут следует заполнять только те поля которые по умолчанию должны быть другими, а не пустыми строками. Все поля у которых значение поля по умолчанию при создании устраивает здесь добавлять не надо. В частности и эти три.

dha01 commented on IS.Mvc/IS.Mvc/Views/Discipline/List.cshtml:
```diff
@@ -0,0 +1,91 @@
+		<div align="center" class="creator">
+			@if (Access.CheckRole("Discipline.Creator"))
+			{
+				@Html.ActionLink("Создать дисциплину", "New", new {id = 10})
```
Что за magick number 10? и каково его назначение тут?

dha01 commented on SQL/Task_80/Insert into Access.role values creator,deleter,updater,reader.sql:
```diff
@@ -0,0 +1,10 @@
+           (code
+           ,mem)
+     values
+           ('Discipline.Creator','Позволяет создать дисциплину'),
```
Пробелы...

dha01 commented on IS.Mvc/IS.Mvc/Views/Discipline/New.cshtml:
```diff
@@ -0,0 +1,61 @@
+	        <th style="width: 100px">
+	            Полное название
+	        </th>
+	        <td>
```
Много пробелов...

DenNikonov committed:
ПЕ-21б Никонов Д.О: Создать интерфейс для работы с дисциплинами

Исправлена табуляция
Исправлены ошибки в контроллере
